utils/funcexp/func_insert.cpp

```diff
@@ -83,14 +84,12 @@ std::string Func_insert::getStrVal(rowgroup::Row& row,
     int64_t strLen = cs->numchars(pos, end);
     
     // Return the original string if start isn't within the string.
-    if ((start < 1) || start >= strLen)
+    if ((start < 0) || start > strLen)
```

Comment:
Since start is now 0 based here, the 2nd part of the || should be `start >= strLen`